### PR TITLE
Add platforms, architectures, and amd64 container build to goreleaser

### LINKED ISSUE #40: Alignment of release binaries platforms
**What is the problem you're trying to solve**

The current release process only produces binaries for amd64 linux and osx. Align release with multi arch binaries in #39 

**Describe the solution you'd like**
Update goreleaser to account for additional binary targets

### PR DESCRIPTION
**Pull request by usrbinkat**
this enhances goreleaser with additional platforms and architectures, as well as an amd64 container build.

closes #40

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,6 +11,9 @@ on:
       - main
       - 'release/**'
 
+permissions:
+  contents: write
```
Comment (sabre1041) [inline .github/workflows/ci.yml:15]: Needs `id-token: write` for keyless signing

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -43,47 +46,75 @@ jobs:
...
       with:
         go-version: 1.17
-    - name: release with goreleaser
+    -
+      name: Run GoReleaser
       id: goreleaser
       uses: goreleaser/goreleaser-action@v2
       with:
-        version: latest
+        distribution: goreleaser
```
Comment (sabre1041) [inline .github/workflows/ci.yml:116]: Receiving errors when testing 

https://github.com/sabre1041/uor-client/runs/7400218665?check_suite_focus=true

**Review by sabre1041 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (vsoch) [reply .github/workflows/ci.yml:116]: I'm not super familiar, but it looks like the build arg `TARGETARCH` isn't defined for this stage: https://github.com/sabre1041/uor-client/blob/396a922199308321b674a2dbf25ad2a37730956c/Containerfile#L16. 

So just add that variable to wherever the build is happening! (I can look more closely after dinner, kind of curious how this Containerfile works)!

Comment (vsoch) [reply .github/workflows/ci.yml:116]: I'm thinking either you can add (or hardcode it if it's just one target for in the container) in this section here, which is what I think drives the buildx build during the gorelease action?

https://github.com/sabre1041/uor-client/blob/396a922199308321b674a2dbf25ad2a37730956c/.goreleaser.yaml#L81-L95

or don't do that, but set a default in the Containerfile for it.

**Comment by vsoch** [comment]
Sorry I should have been more helpful! I think ARG values are only active/relevant for the stage they are defined in.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (usrbinkat) [reply .github/workflows/ci.yml:117]: The variable [`TARGETARCH`](https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope) ([hyperlink](https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope)) is a global variable in docker buildx builds which is sourced from the explicit or implicit platform flag `--platform=linux/amd64`. 

I am relying on this behavior as groundwork to support multi-arch container builds in subsequent PRs from this single Containerfile.

This particular Containerfile design is written to achieve a minimalist / nearly distroless container end result while still being built exclusively from UBI rpm sources.

I seem to have inadvertently broken the Containerfile by moving the `ARG TARGETARCH` line above the second `FROM` line while doing some 'house keeping' in the containerfile.

Comment (vsoch) [reply .github/workflows/ci.yml:117]: Gotcha! And totally OK, those kinds of things happen once in a while. I’ve actually never worked with the Go release action or seen a containerfile so I’m enjoying learning about it! I should give this a whirl for some of my Go projects.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (usrbinkat) [reply .github/workflows/ci.yml:15]: added

File: .github/workflows/ci.yml (hunk 2 above)
Comment (usrbinkat) [reply .github/workflows/ci.yml:117]: goreleaser is new to me too, it's been a fun puzzle to play with :)

**Comment by vsoch** [comment]
Did you mean to close? Are you going to open another one?

**Comment by usrbinkat** [comment]
closed in favor of https://github.com/uor-framework/uor-client-go/pull/49

**Pull request closed without merging**